go/vt/vtgate/planbuilder/rewrite.go

```diff
@@ -112,18 +113,58 @@ func queryRewrite(ctx planningContext, statement sqlparser.SelectStatement) erro
 		semTable:     ctx.semTable,
 		reservedVars: ctx.reservedVars,
 	}
-	sqlparser.Rewrite(statement, r.rewrite, nil)
+	sqlparser.Rewrite(statement, r.rewriteDown, r.rewriteUp)
 	return nil
 }
 
-func (r *rewriter) rewrite(cursor *sqlparser.Cursor) bool {
+func (r *rewriter) rewriteDown(cursor *sqlparser.Cursor) bool {
 	switch node := cursor.Node().(type) {
 	case *sqlparser.Select:
 		rewriteHavingClause(node)
 	case *sqlparser.ExistsExpr:
 		return r.rewriteExistsSubquery(cursor, node)
 	case *sqlparser.Subquery:
+		r.isInSubquery++
 		rewriteSubquery(cursor, r, node)
+	case *sqlparser.AliasedTableExpr:
+		if r.isInSubquery == 0 {
```

Comment:
nit: squash this and the next if into a single one with `||` between

Reply:
done